File: src/main/java/no/nav/syfo/domain/GodkjentPlan.java
package no.nav.syfo.domain;

import lombok.Data;
import lombok.experimental.Accessors;

import java.time.LocalDateTime;
import java.util.Optional;

import static java.util.Optional.empty;

@Data
@Accessors(fluent = true)
public class GodkjentPlan {
    public long id;
    public long oppfoelgingsdialogId;
    public LocalDateTime opprettetTidspunkt;
    public Gyldighetstidspunkt gyldighetstidspunkt;
    public boolean tvungenGodkjenning;
    public LocalDateTime deltMedNAVTidspunkt;
    public boolean deltMedNAV;
    public boolean deltMedFastlege;
    public LocalDateTime deltMedFastlegeTidspunkt;
    public String dokumentUuid;
    public Optional<Avbruttplan> avbruttPlan = empty();
    public String sakId;
    public String journalpostId;
    public String tildeltEnhet;
}

File: src/main/kotlin/no/nav/syfo/api/v3/domain/oppfolgingsplan/GodkjentPlan.kt
package no.nav.syfo.api.v3.domain.oppfolgingsplan

import java.time.LocalDateTime

data class GodkjentPlan(
    val opprettetTidspunkt: LocalDateTime,
    val gyldighetstidspunkt: Gyldighetstidspunkt,
    val tvungenGodkjenning: Boolean,
    val deltMedNAVTidspunkt: LocalDateTime,
    val deltMedNAV: Boolean,
    val deltMedFastlegeTidspunkt: LocalDateTime,
    val deltMedFastlege: Boolean,
    val dokumentUuid: String,
    val avbruttPlan: AvbruttPlan?,
)
